llvm/test/Transforms/Attributor/remove_out_args.ll

```diff
@@ -0,0 +1,16 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt -S -passes=attributor < %s | FileCheck %s
+
+
```

Comment:
This needs about 100 more tests llvm/test/CodeGen/AMDGPU/rewrite-out-arguments.ll and llvm/test/CodeGen/AMDGPU/rewrite-out-arguments-address-space.ll would be better starting points

Reply:
I will. BTW, I don't have an AMD GPU, I have a nvidia one, should J add the tests in NVPTX?

Reply:
It doesn't matter what GPU you have, it doesn't matter when working on the compiler. NVPTX will be less useful for IR testing since it doesn't use a non-0 address space for stack objects

Reply:
Still need all of these tests, using AMDGPU

Reply:
@arsenm, I will test using AMDGPU, but I don't know why my changes don't change this simple test? Do I need to call the some function somewhere? the optimizer dosn't see my changes or do my changes don't affect the input?

Reply:
This is an externally visible function, so it's not allowed to change the signature. It would need to be be internal or similar linkage, with a call use